org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/highlight/HighlightReconcilingStrategy.java

```diff
@@ -77,6 +80,16 @@ public class HighlightReconcilingStrategy
 	private @Nullable IDocument document;
 	private @Nullable Job highlightJob;
 
+	// Debounce to avoid flooding requests while the user moves the caret/mouse.
+	private static final int HIGHLIGHT_DEBOUNCE_MS = 75;
+
+	// Short-lived cache for highlights per document+normalized offset.
+	private static final DocumentOffsetAsyncCache<List<? extends DocumentHighlight>> HIGHLIGHT_CACHE =
```

Comment:
my last comment on the PR: should not invalidate the cache when the document has changed by checking the modification timestamp?

Reply:
that is what DocumentOffsetAsyncCache does internally or what do you mean?